File: components/events/TweetPostedEvent.ts
import { Environment, fetchQuery, graphql } from "relay-runtime"

import { createEventHook } from "components/EventsProvider"

export interface TweetPostedEvent {
  user_id: string
  item_id: string
  autoposted: boolean
}

const fetchTweetQuery = graphql`
  query TweetPostedEventQuery($id: ID!) {
    node(id: $id) {
      id
      ... on TweetGroup {
        tweets {
          body
          mediaURLs
          postedTweetID
        }
        status
        postedAt
        postAfter
      }
    }
  }
`

export async function handleTweetPosted(
  environment: Environment,
  event: TweetPostedEvent
): Promise<void> {
  await fetchQuery(environment, fetchTweetQuery, { id: event.item_id })
}

export const useTweetPostedEvent = createEventHook(
  "TweetPosted",
  handleTweetPosted
)

File: ios/Courier/Courier/Events/TweetPostedEvent.swift
import Combine
import Relay
import CourierGenerated

struct TweetPostedEvent: Decodable {
    var userID: String
    var itemID: String
    var autoposted: Bool?

    enum CodingKeys: String, CodingKey {
        case userID = "user_id"
        case itemID = "item_id"
        case autoposted = "autoposted"
    }
}

private let query = graphql("""
query TweetPostedEventQuery($id: ID!) {
  tweetGroup(id: $id) {
    id
    tweets {
      body
      mediaURLs
      postedTweetID
    }
    status
    postedAt
    postAfter
  }
}
""")

func handleTweetPosted(_ environment: Environment, _ event: TweetPostedEvent) -> AnyPublisher<Never, Error> {
    environment.fetchQuery(TweetPostedEventQuery(variables: .init(id: event.itemID)))
        .ignoreOutput()
        .eraseToAnyPublisher()
}
